cellprofiler/modules/blobdetection.py

```diff
@@ -0,0 +1,379 @@
+            doc="""
+            Step size for values between "Minimum" and "Maximum".
+            """
+        )
+
+        self.count = cellprofiler.setting.Integer(
+            u"Count",
+            10,
+            doc="""
+            Number of intermediate values between "Minimum" and "Maximum".
+            """
+        )
+
+        self.threshold = cellprofiler.setting.Float(
+            u"Threshold",
+            0.2,
+            doc="""
+            The absolute lower bound for intensity of detectable blobs. Regions with intensity less than this value are
+            ignored. Reduce this value to ignore dimmer blobs.
+            """
+        )
+
+        self.overlap = cellprofiler.setting.Float(
+            u"Overlap",
+            0.5,
+            minval=0.0,
+            maxval=1.0,
+            doc="""
+            If the area of two blobs overlaps by a fraction greater than this value, the smaller blob is removed.
+            """
+        )
+
+        self.scale = cellprofiler.setting.Choice(
+            u"Scale",
+            [
+                u"Linear interpolation",
+                u"Logarithm"
+            ],
+            u"Linear interpolation",
+            doc="""
+            Select the method for determining intermediate values between "Minimum" and "Maximum":
+            <ul>
+                <li>
+                    <i>Linear interpolation</i> (default): Choose "Count" number of intermediate values using linear
+                    interpolation.
+                </li>
+
+                <li>
+                    <i>Logarithm</i>: Choose "Count" number of intermediate values using a base 10 logarithmic scale.
+                </li>
+            </ul>
+            """
+        )
+
+    def settings(self):
+        return [
+            self.x_name,
+            self.y_name,
+            self.operation,
+            self.minimum,
+            self.maximum,
+            self.ratio,
+            self.count,
+            self.threshold,
+            self.overlap,
+            self.scale
+        ]
+
+    def visible_settings(self):
+        settings = [
+            self.x_name,
+            self.y_name,
+            self.operation,
+            self.minimum,
+            self.maximum
+        ]
+
+        if self.operation.value in [u"Determinant of the Hessian (DoH)", u"Laplacian of Gaussian (LoG)"]:
+            settings = settings + [
+                self.count
+            ]
+
+        if self.operation.value == u"Difference of Gaussians (DoG)":
+            settings = settings + [
+                self.ratio
+            ]
+
+        settings = settings + [
+            self.threshold,
+            self.overlap
+        ]
+
+        if self.operation.value in [u"Determinant of the Hessian (DoH)", u"Laplacian of Gaussian (LoG)"]:
+            settings = settings + [
+                self.scale
+            ]
+
+        return settings
+
+    def on_setting_changed(self, setting, pipeline):
```

Comment:
Each operation's `skimage.feature` equivalent has slightly different default parameters. This has the benefit of changing the settings to match those default parameters when the operation is changed. The downside is that it forgets/wipes out user defined settings when the operation is changed.